ai-gateway/custom-providers/azure-openai.mdx

```diff
@@ -0,0 +1,253 @@
+
+```mermaid
+graph LR
+    A[Client] --> B[AI Gateway]
+    B --> C[Azure OpenAI]
+```
+
+## Step 1: Get Azure OpenAI details
+
+From the Azure Portal, gather:
+
+1. **Endpoint URL**: `https://your-resource.openai.azure.com`
+2. **API Key**: From "Keys and Endpoint" section
+3. **Deployment Name**: The name you gave your model deployment
+
+## Step 2: Store your API key
+
+Add your Azure OpenAI API key to [ngrok secrets](/traffic-policy/secrets):
+
+```bash
+ngrok api secrets create \
+  --name azure-openai \
+  --secret-data '{"api-key": "your-azure-api-key"}'
+```
+
+<Tip>
+You can also create secrets in the [ngrok Dashboard](https://dashboard.ngrok.com/vaults).
+</Tip>
+
+## Step 3: Configure the AI gateway
+
+Create your traffic policy:
+
+```yaml
+on_http_request:
+  - type: ai-gateway
+    config:
+      headers:
+        api-version: "2024-02-15-preview"
+      providers:
+        - id: "azure-openai"
+          base_url: "https://your-resource.openai.azure.com/openai/deployments/your-deployment"
+          api_keys:
+            - value: ${secrets.get('azure-openai', 'api-key')}
+          models:
+            - id: "gpt-4o"
+```
+
+<Note>
+Azure OpenAI requires the `api-version` header. The AI Gateway adds this to all requests.
+</Note>
+
+## Step 4: Use with OpenAI SDK
+
+<CodeGroup>
+```python Python
+from openai import OpenAI
+
+client = OpenAI(
+    base_url="https://your-ai-subdomain.ngrok.app/v1",
+    api_key="unused"  # Gateway handles auth
+)
+
+response = client.chat.completions.create(
+    model="azure-openai:gpt-4o",
+    messages=[{"role": "user", "content": "Hello!"}]
+)
+
+print(response.choices[0].message.content)
+```
+
+```typescript TypeScript
+import OpenAI from "openai";
+
+const client = new OpenAI({
+  baseURL: "https://your-ai-subdomain.ngrok.app/v1",
+  apiKey: "unused"
+});
+
+const response = await client.chat.completions.create({
+  model: "azure-openai:gpt-4o",
+  messages: [{ role: "user", content: "Hello!" }]
+});
+
+console.log(response.choices[0].message.content);
+```
+</CodeGroup>
+
+## Advanced configuration
+
+### Multiple deployments
+
+Configure multiple Azure OpenAI deployments:
+
+```yaml
+on_http_request:
+  - type: ai-gateway
+    config:
+      headers:
+        api-version: "2024-02-15-preview"
+      providers:
+        - id: "azure-gpt4"
+          base_url: "https://your-resource.openai.azure.com/openai/deployments/gpt-4o-deployment"
+          api_keys:
+            - value: ${secrets.get('azure-openai', 'api-key')}
+          models:
+            - id: "gpt-4o"
+        
+        - id: "azure-gpt35"
+          base_url: "https://your-resource.openai.azure.com/openai/deployments/gpt-35-deployment"
+          api_keys:
+            - value: ${secrets.get('azure-openai', 'api-key')}
+          models:
+            - id: "gpt-3.5-turbo"
+```
+
+### Multiple regions
+
+Failover across Azure regions:

```

Comment:
These won't always automatically failover unless the clients specifically request like `models: ["azure-eastus:gpt-4o", "azure-westus:gpt-4o"]`, or if the client does just request `gpt-4o` or auto/nothing and the api key strategy selects both, then we will, but if the client specifies the provider, we won't try another unless the client says to.

TL;DR `model: "azure-eastus:gpt-4o"` will never fall back to the other region and users might find that nuance confusing with this example?